runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/environment/DockerEnvironmentFactory.java

```diff
@@ -112,6 +122,10 @@ public RemoteEnvironment createEnvironment(Environment environment) throws Excep
     String workerId = idGenerator.getId();
 
     // Prepare docker invocation.
+    String pathSafeWorkerId =
+        BaseEncoding.base64Url().encode(workerId.getBytes(StandardCharsets.UTF_8));
```

Comment:
Base64 uses slashes in its encoding.

Reply:
Base64-URL should use `_` instead of `/`.

Reply:
Just checked the [RFC](https://tools.ietf.org/html/rfc4648#section-5) to confirm this.